cli/setup/setup.go

```diff
@@ -30,6 +31,12 @@ func Setup(args []string, tempDir string) (_ []*plugin.Plugin, bazelArgs []strin
 		return nil, nil, nil, nil, err
 	}
 
+	// Save some flags from the current invocation for non-Bazel commands such
+	// as `bb ask`.
+	// Args are saved before the sidecar rewrites them as API requests require
+	// the original --bes_backend value.
+	bazelArgs = storage.SaveFlags(bazelArgs)
```

Comment:
An alternative could be to support arbitrary gRPC requests on the sidecar socket. When I tried to connect to it to fetch invocations, I ran into errors, but perhaps I was just holding it wrong.